coderd/notifications/notifications_test.go

```diff
@@ -938,53 +986,306 @@ func TestNotificationTemplates_Golden(t *testing.T) {
+				}
+
+				webhookManager, err := notifications.NewManager(
+					webhookCfg,
+					*db,
+					defaultHelpers(),
+					createMetrics(),
+					logger.Named("manager"),
+				)
+				require.NoError(t, err)
+
+				webhookManager.Run(ctx)
+
+				httpEnqueuer, err := notifications.NewStoreEnqueuer(
+					defaultNotificationsConfig(database.NotificationMethodWebhook),
+					*db,
+					defaultHelpers(),
+					logger.Named("enqueuer"),
+					quartz.NewReal(),
+				)
+				require.NoError(t, err)
+
+				_, err = httpEnqueuer.EnqueueWithData(
+					ctx,
+					user.ID,
+					tc.id,
+					tc.payload.Labels,
+					tc.payload.Data,
+					user.Username,
+					user.ID,
+				)
+				require.NoError(t, err)
+
+				require.Eventually(t, func() bool {
+					return webhookReceived
+				}, testutil.WaitShort, testutil.IntervalFast)
+
+				require.NoError(t, err)
+
+				// Handle the body that was read in the http server here.
+				// We need to do it here because we can't call require.* in a separate goroutine, such as the http server handler
+				require.NoError(t, readErr)
+				var prettyJSON bytes.Buffer
+				err = json.Indent(&prettyJSON, body, "", "  ")
+				require.NoError(t, err)
+
+				content := normalizeGoldenWebhook(prettyJSON.Bytes())
+
+				partialName := strings.Split(t.Name(), "/")[1]
+				goldenFile := filepath.Join("testdata", "rendered-templates", "webhook", partialName+".json.golden")
+				if *updateGoldenFiles {
+					err = os.MkdirAll(filepath.Dir(goldenFile), 0o755)
+					require.NoError(t, err, "want no error creating golden file directory")
+					err = os.WriteFile(goldenFile, content, 0o600)
+					require.NoError(t, err, "want no error writing body golden file")
+					return
+				}
+
+				wantBody, err := os.ReadFile(goldenFile)
+				require.NoError(t, err, fmt.Sprintf("missing golden notification body file. %s", hint))
+				require.Equal(t, wantBody, content, fmt.Sprintf("smtp notification does not match golden file. If this is expected, %s", hint))
+			})
 		})
 	}
 }
 
+func normalizeGoldenEmail(content []byte) []byte {
+	const (
+		constantDate      = "Fri, 11 Oct 2024 09:03:06 +0000"
+		constantMessageID = "02ee4935-73be-4fa1-a290-ff9999026b13@blush-whale-48"
+		constantBoundary  = "bbe61b741255b6098bb6b3c1f41b885773df633cb18d2a3002b68e4bc9c4"
+	)
+
+	dateRegex := regexp.MustCompile(`Date: .+`)
+	messageIDRegex := regexp.MustCompile(`Message-Id: .+`)
+	boundaryRegex := regexp.MustCompile(`boundary=([0-9a-zA-Z]+)`)
+	submatches := boundaryRegex.FindSubmatch(content)
+	if len(submatches) == 0 {
+		return content
+	}
+
+	boundary := submatches[1]
+
+	content = dateRegex.ReplaceAll(content, []byte("Date: "+constantDate))
+	content = messageIDRegex.ReplaceAll(content, []byte("Message-Id: "+constantMessageID))
+	content = bytes.ReplaceAll(content, boundary, []byte(constantBoundary))
+
+	return content
+}
+
+func normalizeGoldenWebhook(content []byte) []byte {
+	const constantUUID = "00000000-0000-0000-0000-000000000000"
```

Comment:
```
{
  "_version": "1.1",
  "msg_id": "00000000-0000-0000-0000-000000000000",
  "payload": {
    "_version": "1.1",
    "notification_name": "Workspace Autobuild Failed",
    "notification_template_id": "00000000-0000-0000-0000-000000000000",
    "user_id": "00000000-0000-0000-0000-000000000000",
```

Ok, I know why you did this but I'm wondering if there is a way to keep stable uuids written to the golden file 🤔. It could help find out why some labels change, and prevent bugs in the future.

PS. If this is difficult->impossible, it can be postponed for later (add FIXME label).

Reply:
I mean, we could replace dynamic UUIDs with some const/known values rather than zeros. For instance:

user_id -> `ceb6824e-7fe1-4000-aad9-b80bb141560a`

Reply:
ACK. Thanks